ee/hogai/django_checkpoint/checkpointer.py

```diff
@@ -124,23 +123,25 @@ def list(
         if limit:
             qs = qs[:limit]
 
-        for checkpoint in qs:
+        async for checkpoint in qs:
```

Comment:
As I understand, you first query to get all the checkpoints based on criteria, then N `pending_sends` queries and N `pending_writes` queries. 

Would something like `.prefetch_related("writes", "pending_sends")` be useful to add to the checkpoint query?

Reply:
It's rather a question-suggestion, as I see DB models like `ConversationCheckpointWrite`, so not sure how optimized the approach is by default.